netket/hilbert/particle.py

```diff
@@ -41,52 +43,80 @@ def __init__(
                 if there are no boundaries?). If a finite value is given, a minimum value of zero is assumed for the
                 quantum number(s).
                 A particle in a 3D box of size L would take `(L,L,L)`. A rotor model would take e.g. `(2pi,)`.
-            pbc: Tuple or bool indicating whether to use periodic boundary conditions in a given physical dimension.
-                If tuple it must have the same length as domain. If bool the same value is used for all the dimensions
-                defined in domain.
+            pbc: Tuple of bools or bool indicating whether to use periodic boundary conditions in a given physical
```

Comment:
personally, I would be in favor of breaking the interface instead of adding deprecations and convoluted consutrctors etc... but OK, it's not so user friendly maybe :)